crates/shared/src/sources/balancer_v2/pools.rs

```diff
@@ -100,6 +102,10 @@ pub trait FactoryIndexing: Send + Sync + 'static {
     /// Additionally, a block spec and a batch call context is passed in to
     /// specify exactly the block number the state should be read for, and allow
     /// for more optimal performance when fetching a large number of pools.
+    ///
+    /// This method should return `None` if the pool is disabled. This allows
```

Comment:
Isn't disabled similar to paused in that the pool could later become enabled? Why are we returning PoolState for paused pools but not for disabled ones? If we ignore disabled pools here will we still be able to fetch them later on without a restart of the backend?

Or can we return None if `common::PoolState` is not active - this might save some unnecessary queries for pools we don't use later anyways?

Reply:
So, from what I understand LBPs have separate `paused` and `enabled` states. Specifically, the `paused` state is shared by all pools (its part of the Balancer `BasePool` implementation, while `swapEnabled` is specific to LBPs).

The idea behind behind this change is that by lowering the `swapEnabled` check into the factory indexing implementation, we allow arbitrary logic for determining if a pool is tradable or not specific to the factory.

> Why are we returning PoolState for paused pools but not for disabled ones? 

Hmm, a bit of a code smell 😞. The idea is that the `paused` state is checked by the pool registry so the individual factory indexing implementations don't need to care about it. I agree that its a bit ambiguous from the `FactoryIndexing` trait perspective where its unclear what the caller should do for paused pools. For now, I'll add a comment and try to make a small PR to change this around a bit to get rid of this code smell.

> If we ignore disabled pools here will we still be able to fetch them later on without a restart of the backend?

No, this is used for querying the current pool state, not for registering the pool as a "known pool". So we will actually re-query the pool for every batch and check the `swapEnabled` then (from what I understand, this is used so a project can create and fund a pool and then have it become enabled at a later point in time, so we do need to re-query this). 

> this might save some unnecessary queries for pools we don't use later anyways?

It does. We currently prefer batching instead of first querying all "enabled states" and then the rest of the pool state (i.e. balances, weights, etc.), the down side being some additional pressure on the node in cases where the pool is disabled. Last I checked, there are no `paused` pools, so this is definitely the right choice for paused pools - we can keep an eye on LBPs to see if we are eagerly querying too much data.